pkg/kp/pcstore/pcstoretest/fake_pcstore.go

```diff
@@ -56,3 +57,19 @@ func (p *FakePCStore) Delete(id fields.ID) error {
 	delete(p.podClusters, id)
 	return nil
 }
+
+func (p *FakePCStore) FindWhereLabeled(
```

Comment:
might be worth putting a `labels.NewFakeApplicator()` onto the fake store and actually write/delete/query labels so if the label set ever diverges from the fields the behavior will stay correct, but not necessary today.